### PR TITLE
Refactor get_sensor_data helper method in #722

### PR DESCRIPTION
**Pull request by eitoball**

### CODE REVIEW

File: app/helpers/device_stories_helper.rb [hunk 1]
```diff
@@ -72,4 +72,34 @@ def last_air_quality_values(last_values)
...
+    'air_sensors' => %w(pms_pm10_0 pms_pm02_5 pms_pm01_0),
+    'bat_voltage' => ['bat_voltage'],
+    'temperature_C' => ['temperature_C'],
+    'temperature_F' => ['temperature_F'],
+    'humidity' => ['humidity'],
+    'pressure' => ['pressure'],
+    'charging' => ['charging']
+  }.freeze
+
+  def sensor_data_by_sensors(sensor_data)
```
Comment (eitoball) [inline app/helpers/device_stories_helper.rb:87]: This is a refactored method of `get_sensor_data`. This method expects `sensor_data`; data from `IngestMeasurement.query_sensor_data`.

File: app/helpers/device_stories_helper.rb [hunk 2]
```diff
@@ -72,4 +72,34 @@ def last_air_quality_values(last_values)
...
+    CATEGORY_SENSOR_NAMES.transform_values do |names|
+      names.each_with_object([]) do |name, arr|
+        arr << { name: name, value: data_by_dates[name] } if data_by_dates.key?(name)
+      end
+    end
+  end
+
+  def convert_by_dates(buckets)
+    buckets.each_with_object(Hash.new { |h, k| h[k] = {} }) do |bucket, hsh|
+      date = bucket['key_as_string']
```
Comment (eitoball) [inline app/helpers/device_stories_helper.rb:98]: This line expects data from 'IngestMeasurement.query_sensor_data` in that `key_as_string` is formatted in `yyyy-MM-dd HH`.

**Comment by matschaffer** [comment]
Nice looking factor @eitoball !

**Comment by ma2gedev** [comment]
LGTM! It's much simpler than before. 🆒

**Review by ma2gedev — APPROVED** [review]

File: app/helpers/device_stories_helper.rb [hunk 3]
```diff
@@ -74,43 +74,38 @@ def last_air_quality_values(last_values)
...
-        sensor_exists = false
-        q.response['aggregations']['sensor_data']['buckets'].each do |bucket|
-          date = Time.at(bucket['key'] / 1000.0).strftime('%Y-%m-%d %H')
-          sensor_value = bucket[sensor]['value']
-          if sensor_value then sensor_exists = true end
-          hash_sensor.merge!(date => sensor_value)
-        end
-        if sensor_exists then sensor_type_hashes.push({ "name": sensor, "data": hash_sensor }) end
+  def sensor_data_by_sensors(sensor_data)
+    data_by_dates = convert_by_dates(sensor_data.dig('aggregations', 'sensor_data', 'buckets') || [])
```
Comment (matschaffer) [inline app/helpers/device_stories_helper.rb:89]: I put this on dev but it's getting `undefined method `dig' for #<Elasticsearch::Model::Response::Response:0x0000000007d4ee58>`

https://5bc3d4f7330a4459881197a85659caf5.us-west-2.aws.found.io:9243/app/apm#/services/safecastapi-dev/errors/1baaae338359071560c867ba08f88b5c?rangeFrom=now-15m&rangeTo=now&refreshPaused=true&refreshInterval=0

**Pull request merged**